pandas/core/reshape.py

```diff
@@ -771,6 +771,8 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
     if value_vars is not None:
         if not isinstance(value_vars, (tuple, list, np.ndarray)):
```

Comment:
this checking (and in the clause above) should be ``not is_list_like(value_vars)``